app/storage/knowledge-document-repo.js

```diff
@@ -25,7 +25,15 @@ const listKnowledge = async (category = '', orderBy = 'lastModified ', orderByDi
     const metadata = mapMetadataToBase(blob.metadata)
 
     blob.metadata = metadata
-    if (category === '' || metadata.category === category) {
+    if (search !== '' && metadata.fileName) {
+      if (metadata.fileName.toLowerCase().indexOf(search.toLowerCase()) > -1) {
```

Comment:
Could you split this up a bit? Maybe create a new var to store the lower case file name?